packages/neon-invoker/src/index.ts

```diff
@@ -78,41 +86,73 @@ export class NeonInvoker implements Neo3Invoker {
 
     const currentHeight = await rpcClient.getBlockCount()
 
-    const trx = new tx.Transaction({
-      script: u.HexString.fromHex(script),
-      validUntilBlock: currentHeight + 100,
-      signers: NeonInvoker.buildMultipleSigner(this.account, cim.signers),
+    const trx = NeonInvoker.buildTransaction({
+      account: this.account,
+      script,
+      signers: cim.signers,
+      validUntilBlock: currentHeight + 100
     })
 
     const config = {
       ...this.rpcConfig,
       account: this.account,
     }
 
+    const systemFeeOverride = await NeonInvoker.overrideSystemFeeOnTransaction(trx, config, cim)
+
+    const networkFeeOverride = await NeonInvoker.overrideNetworkFeeOnTransaction(trx, config, cim, this.account)
+
+    await NeonInvoker.addFeesToTransaction(trx, {
+      ...config,
+      systemFeeOverride,
+      networkFeeOverride,
+    })
+
+    this.signTransaction(trx)
+
+    return await this.sendTransaction(trx)
+  }
+
+  static buildTransaction({ script, validUntilBlock, account, signers }: BuildTransaction) {
+    return new tx.Transaction({
+      script: u.HexString.fromHex(script),
+      validUntilBlock,
+      signers: NeonInvoker.buildMultipleSigner(account, signers)
+    })
+  }
+
+  static async overrideSystemFeeOnTransaction(trx: Neon.tx.Transaction, config: CommonConfig, cim: ContractInvocationMulti) {
     const systemFeeOverride = cim.systemFeeOverride
       ? u.BigInteger.fromNumber(cim.systemFeeOverride)
       : cim.extraSystemFee
         ? (await experimental.txHelpers.getSystemFee(trx.script, config, trx.signers)).add(cim.extraSystemFee)
         : undefined
+    return systemFeeOverride
+  }
 
+  static async overrideNetworkFeeOnTransaction(trx: Neon.tx.Transaction, config: CommonConfig, cim: ContractInvocationMulti, account: wallet.Account) {
     const networkFeeOverride = cim.networkFeeOverride
       ? u.BigInteger.fromNumber(cim.networkFeeOverride)
       : cim.extraNetworkFee
-        ? (await experimental.txHelpers.calculateNetworkFee(trx, this.account, config)).add(cim.extraNetworkFee)
+        ? (await experimental.txHelpers.calculateNetworkFee(trx, account, config)).add(cim.extraNetworkFee)
```

Comment:
I think we can use this.account

Reply:
ok, make a sense I will change

Reply:
so you think that this method doesn't need be static?

Reply:
yes, to be able to use this.account instead of passing the account as parameter